src/NodeosActionReader.ts

```diff
@@ -78,4 +78,19 @@ export class NodeosActionReader extends AbstractActionReader {
       throw new RetrieveBlockError()
     }
   }
+
+  protected async setup(): Promise<void> {
+    if (this.initialized) {
+      return
+    }
+
+    try {
+      await request.get({
+        url: `${this.nodeosEndpoint}/v1/chain/get_info`,
+        json: true,
+      })
+    } catch (err) {
+      throw new NotInitializedError('Cannot reach supplied nodeos endpoint.')
```

Comment:
Maybe we shouldn't swallow the underlying error here for better debugging.

Reply:
I thought about that, but then the client code can't do initialization specific logic solely based on `NotInitializedErrors`. That may be okay, in which case I can throw the underlying error, but that's why I did it this way

Reply:
So, if they wrap `initialize` in a try/catch, then any error that they catch can be assumed to be an initialization error, right?

Reply:
The initial thought I had was you could do specific `typeof NotInitializedError` stuff if you want, but yeah there's no reason it couldn't throw other errors and those would still be caught by a generic `try/catch`.